src/components/ArcSlider.tsx

```diff
@@ -0,0 +1,249 @@
+		document.addEventListener('touchend', this.handleMouseUp);
+		document.addEventListener('touchcancel', this.handleMouseUp);
+	}
+
+	componentWillUnmount() {
+		document.removeEventListener('mousemove', this.handleMouseMove);
+		document.removeEventListener('touchmove', this.handleMouseMove);
+		document.removeEventListener('mouseup', this.handleMouseUp);
+		document.removeEventListener('touchend', this.handleMouseUp);
+		document.removeEventListener('touchcancel', this.handleMouseUp);
+	}
+
+	handleMouseUp = () => {
+		this.setState({ isDragging: false });
+	};
+
+	handleMouseDown = (e: any) => {
+		e.preventDefault();
+		this.setState({ isDragging: true });
+	};
+
+	handleMouseMove = (e: any) => {
+		if (this.state.isDragging && this.$slider) {
+			const { left, top, width, height } = this.$slider.getBoundingClientRect();
+
+			const sliderCenter = {
+				x: left + width / 2,
+				y: top + height / 2,
+			};
+
+			let mouseLocation = {
+				x: e.pageX,
+				y: e.pageY,
+			};
+
+			// If this is a touch event, the page coordinates need to be retrieved
+			// from the `touches` collection
+			if (!!e.touches) {
+				mouseLocation = {
+					x: e.touches[0].pageX,
+					y: e.touches[0].pageY,
+				};
+			}
+
+			const { angleRange } = this;
+
+			// Normalize the angle to make 0 degrees the topmost point of the arc
+			let angleDegrees =
+				getAngleBetweenPoints(sliderCenter, mouseLocation) + 90;
+
+			if (angleDegrees > 180) {
+				angleDegrees = Math.max(angleDegrees - 360, -angleRange);
+			} else if (angleDegrees < -angleRange) {
+				angleDegrees = -angleRange;
+			} else if (angleDegrees > angleRange) {
+				angleDegrees = angleRange;
+			}
+
+			const value = (angleDegrees + angleRange) / (angleRange * 2);
+
+			this.setState({
+				innerValue: value,
+			});
+
+			if (this.props.onValueChange) {
+				this.props.onValueChange(value);
+			}
+		}
+	};
+
+	render() {
+		const { angleRange } = this;
+		const { innerValue } = this.state;
+
+		// If there is no value prop, use the innerValue so that the slider can
+		// behave like an uncontrolled input
+		const value =
+			this.props.value == null
+				? innerValue
+				: constrainNumber(this.props.value, 0, 1);
+		const { onValueChange, children, ...props } = this.props;
+		const color = this.props.fillColor || '#FEC400';
+		const background = this.props.background || '#707274';
+		const valueAngle = angleRange * 2 * value - angleRange;
+
+		return (
+			<ArcSliderContainer {...props}>
+				<StyledSVG
+					style={{
+						cursor: this.state.isDragging ? 'grabbing' : 'grab',
+					}}
+					innerRef={(elem: any) => (this.$slider = elem)}
+					viewBox="0 0 600 600"
+					onMouseDown={this.handleMouseDown}
+					onTouchStart={this.handleMouseDown}
+				>
+					<defs>
+						<linearGradient
+							id="rendition-arc-slider-gradient"
+							x1="0%"
+							y1="0%"
+							x2="100%"
+							y2="0%"
+						>
+							<stop offset="0%" stop-color={darken(color)} />
```

Comment:
I was getting a warning in the console about this.
> Invalid DOM property `stop-color`. Did you mean `stopColor`?

Do we need to change this?

Reply:
Yeah I think so